### PR TITLE
Reorganize the installation documents.

### PR DESCRIPTION
**Pull request by trivialfis**
* Split up installation and building from source.
* Use consistent section titles.
* Documents from jvm package are gathered into the installation guide and building from source guide.

The installation doc is split into 2 sections, one for stable release another for the nightly build.  Each section contains subsections for Python, R, and JVM.

Also building from source code is now an independent page detailing the instructions for each package.

It's nowhere close to what https://github.com/dmlc/xgboost/issues/5282 is describing, but I think at least it cleanups the current document a little bit.

Here are the TOCs for installation and build:
* Install
install

* Build
build

### CODE REVIEW

File: doc/build.rst [hunk 1]
```diff
@@ -502,20 +442,76 @@ If on Windows you get a "permission denied" error when trying to write to ...Pro
...
-Trouble Shooting
-================
 
-1. Compile failed after ``git pull``
+*********************
+Building JVM Packages
+*********************
 
-   Please first update the submodules, clean all and recompile:
+Building XGBoost4J using Maven requires Maven 3 or newer, Java 7+ and CMake 3.13+ for compiling the JNI bindings.
```
Comment (hcho3) [inline doc/build.rst:450]:
```suggestion
Building XGBoost4J using Maven requires Maven 3 or newer, Java 7+ and CMake 3.13+ for compiling Java code as well as the Java Native Interface (JNI) bindings.
```

File: doc/install.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
+##################
+Installation Guide
+##################
+
+XGBoost provides binary packages for some language bindings.  The binary packages support
+the GPU algorithm (``gpu_hist``) on machines with NVIDIA GPUs. Please note that **training
+with multiple GPUs is only supported for Linux platform**. See :doc:`gpu/index`.  Also we
+have both stable releases and nightly builds, see below for how to install them.  For
+building from source, checkout :doc:`here </build>`.
```
Comment (hcho3) [inline doc/install.rst:9]:
```suggestion
building from source, visit :doc:`this page </build>`.
```

File: doc/install.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+.. contents:: Contents
+
+Stable Release
+==============
+
+Python
+------
+
+Pre-built binary are uploaded to Pypi for each release.  Supported platforms are Linux
+with x86_64, aarch, Windows x86_64 and OSX.
```
Comment (hcho3) [inline doc/install.rst:20]:
```suggestion
Pre-built binary are uploaded to PyPI (Python Package Index) for each release.  Supported platforms are Linux (x86_64, aarch64), Windows (x86_64) and MacOS (x86_64).
```

File: doc/install.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+.. |cross| unicode:: U+2718
+
++-------------------+---------+----------------------+
+| Platform          | GPU     | Multi-Node-Multi-GPU |
++===================+=========+======================+
+| Linux x86_64      | |tick|  |  |tick|              |
++-------------------+---------+----------------------+
+| Linux aarch64     | |cross| |  |cross|             |
++-------------------+---------+----------------------+
+| OSX               | |cross| |  |cross|             |
```
Comment (hcho3) [inline doc/install.rst:40]:
```suggestion
| MacOS             | |cross| |  |cross|             |
```

File: doc/install.rst [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+     single CPU core, leading to suboptimal training speed.
+
+* We also provide **experimental** pre-built binary on Linux x86_64 with GPU support.
+  Download the binary package from the Releases page. The file name will be of the form
+  ``xgboost_r_gpu_linux_[version].tar.gz``. Then install XGBoost by running:
+
+  .. code-block:: bash
+
+    # Install dependencies
+    R -q -e "install.packages(c('data.table', 'magrittr', 'jsonlite', 'remotes'))"
```
Comment (hcho3) [inline doc/install.rst:72]:
```suggestion
    R -q -e "install.packages(c('data.table', 'magrittr', 'jsonlite'))"
```

It turns out that `remotes` is not necessary.

File: doc/install.rst [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+Nightly Build
+=============
+
+
+Python
+------
+
+Nightly builds are available. You can go to `this page <https://s3-us-west-2.amazonaws.com/xgboost-nightly-builds/list.html>`_,
+find the wheel with the commit id you want and install it with pip:
```
Comment (hcho3) [inline doc/install.rst:132]:
```suggestion
find the wheel with the commit ID you want and install it with pip:
```

**Review by hcho3 — CHANGES_REQUESTED** [review]
Nicely done! I like how the new installation doc is organized. I have some minor comments.

**Comment by trivialfis** [comment]
Thanks for the suggestions, all applied.

File: doc/build.rst [hunk 7]
```diff
@@ -56,10 +57,10 @@ on the binding you choose).  The minimal building requirement is
 For a list of CMake options like GPU support, see ``#-- Options`` in CMakeLists.txt on top
 level of source tree.
 
-Building on Linux distributions
+Building on Linux and other UNIX-like systems
 ===============================
```
Comment (hcho3) [inline doc/build.rst:61]:
```suggestion
Building on Linux and other UNIX-like systems
=============================================
```

File: doc/build.rst [hunk 8]
```diff
@@ -69,10 +70,10 @@ On Ubuntu, after obtaining the source code, one builds XGBoost by running CMake:
   cmake ..
   make -j$(nproc)
 
-Building on OSX
+Building on MacOS
 ===============
```
Comment (hcho3) [inline doc/build.rst:74]:
```suggestion
Building on MacOS
=================
```

**Review by hcho3 — APPROVED** [review]

**Pull request merged**